CRM/Custom/Form/Field.php

```diff
@@ -613,6 +621,12 @@ public static function formRule($fields, $files, $self) {
       }
     }
 
+    if ($dataType === 'EntityReference') {
+      if (empty($fields['fk_entity'])) {
+        $errors['fk_entity'] = ts('Selecting an entity is required');
```

Comment:
Advocating for *entity type*, as it's not the entity itself that is to be selected here (although it in fact is: the API entity, but that confuses people, I think).

Reply:
Sure we can change the label. I think the machine name `fk_entity` is appropriate though.

Reply:
I was actually just thinking of the label, so :+1: for keeping the internal field name.

Reply:
Ah - the `fk_entity` came up here too. Yeah does this field actually hold the `entity_type` - I was assuming from my read above it would hold an id. So yeah - if that is what it means I would have gone with a machine name that includes the word type (reference_entity_type)

Reply:
`fk_entity` is used a lot in API4 code to store the API4 entity name (i.e. the short BAO entity name, i.e. the type of entities being referenceable). Using a separate naming scheme here doesn't seem necessary to me.

But I think it adds to the confusion that API *entities* are mapped to BAO *entity types* and there is no clear semantic distinction between an *entity type* (i.e. a *class* of things) and an *entity* (i.e. a particular instance of that *class*) in CiviCRM in general.